alfred-telemetry-platform/src/main/java/eu/xenit/alfred/telemetry/binder/NodeMetrics.java

```diff
@@ -0,0 +1,81 @@
+package eu.xenit.alfred.telemetry.binder;
+
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.MeterRegistry;
+
+import java.util.concurrent.TimeUnit;
+import java.util.function.ToDoubleFunction;
+
+import io.micrometer.core.instrument.TimeGauge;
+import org.alfresco.repo.domain.node.NodeDAO;
+import org.alfresco.repo.domain.permissions.AclDAO;
+import org.alfresco.repo.transaction.RetryingTransactionHelper;
+import org.alfresco.service.transaction.TransactionService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class NodeMetrics {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(NodeMetrics.class);
+
+    public static final String ALC_PREFIX = "alfresco.alc";
+    public static final String DAO_PREFIX = "alfresco.node";
+    private TransactionService transactionService;
+    private TimeUnit baseTimeUnit = TimeUnit.MILLISECONDS;
+
+    private NodeDAO nodeDAO;
+    private AclDAO aclDAO;
+    private MeterRegistry registry;
+
+    public NodeMetrics(NodeDAO nodeDAOComponent, AclDAO aclDAOComponent, TransactionService transactionService,
+                       MeterRegistry registry, boolean enabled) {
+        this.nodeDAO = nodeDAOComponent;
+        this.aclDAO = aclDAOComponent;
+        this.transactionService = transactionService;
+        this.registry = registry;
+        if (enabled) {
+            registerMetrics();
+        }
+    }
+
+    private void registerMetrics() {
+        LOGGER.info("Registering Solr metrics");
+
+        Gauge.builder(String.format("%s.%s", DAO_PREFIX, "maxTxNode"), nodeDAO, wrapDAOFunction(NodeDAO::getMaxNodeId))
+                .baseUnit("number")
+                .register(registry);
+        Gauge.builder(String.format("%s.%s", DAO_PREFIX, "maxTxnId"), nodeDAO, wrapDAOFunction(NodeDAO::getMaxTxnId))
+                .baseUnit("number")
+                .register(registry);
+        TimeGauge.builder(String.format("%s.%s", DAO_PREFIX, "maxTxnCommitTime"), nodeDAO, baseTimeUnit, wrapDAOFunction(NodeDAO::getMaxTxnCommitTime))
+                .register(registry);
+        Gauge.builder(String.format("%s.%s", ALC_PREFIX, "maxChangeSetId"), aclDAO, wrapAlcFunction((dummy) -> {
+            long maxCommitTime = System.currentTimeMillis()+1L;
```

Comment:
Do you trust the local time?

In some situations where time is not synchronized between database and alfresco this may result in wrong metrics.

Using the `maxChangeSetCommitTime` may return a better result?

Reply:
That's what I first used. I'll revert it back.